compiler/shared/main/scala/mlscript/compiler/printer/ExprPrinter.scala

```diff
@@ -0,0 +1,110 @@
+package mlscript.compiler.printer
+
+import mlscript.compiler.{Expr, Isolation, Item, Module, Parameter}
+
+class ExprPrinter:
+  private val printer = BlockPrinter()
+
+  import printer.{endLine, enter, leave, print}
+
+  private def show(module: Module): Unit = module.items.foreach {
+    case expr: Expr => show(expr)
+    case item: Item => show(item)
+  }
+
+  private def show(params: List[Parameter]): String =
+    params.iterator.map {
+      case (spec, Expr.Ref(name)) => (if spec then "#" else "") + name
+    }.mkString("(", ", ", ")")
+
+  private def show(item: Item): Unit = item match
+    case Item.TypeDecl(Expr.Ref(name), kind, typeParams, params, parents, body) =>
+      val typeParamsStr = if typeParams.isEmpty then ""
+        else typeParams.iterator.map(_.name).mkString("[", ", ", "]")
+      val reprParents = if parents.isEmpty then ""
+        else parents.iterator.map { case (parent, args) =>
+          parent.show(true) + args.iterator.mkString("(", ", ", ")")
+        }.mkString(": ", ", ", "")
+      print(s"$kind $name$typeParamsStr${show(params)}$reprParents ")
+      show(body)
+    case Item.FuncDecl(Expr.Ref(name), params, body) =>
+      print(s"fun $name${show(params)} =")
+      enter()
+      show(body)
+      leave()
+    case Item.FuncDefn(Expr.Ref(name), typeParams, polyType) =>
+      val reprTypeParams = if typeParams.isEmpty then "" else
+        s"${typeParams.mkString("[", ", ", "]")} => "
+      print(s"fun $name: $reprTypeParams${polyType.show}")
+
+  private def show(isolation: Isolation): Unit =
+    enter("{", "}")
+    isolation.items.foreach {
+      case expr: Expr => show(expr)
+      case item: Item => show(item)
+    }
+    leave()
+
+  private def show(expr: Expr) =
+    print(expr.toString)
+    endLine()
+
+  def toLines: List[String] = printer.toLines
+
+  override def toString(): String = printer.toString
+
+  // private def show(expr: Expr): Unit = expr match
```

Comment:
Can we remove this commented code?

Reply:
Yes. It has been moved to `Expr::toString`